### PR TITLE
core: Barrier

### PR DESCRIPTION
**Pull request by fwbrasil**
In honor of the 500th person to star the repository 🎉 Thank you @wendersonferreira!!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2286,7 +2286,7 @@ val d: Boolean < IO =
     a.map(_.isDone)
 ```
 
-### Latch: Fiber Coordination
+### Latch: Countdown Synchronization
```
Comment (fwbrasil) [inline README.md:2289]: More specific since `Barrier` is similar.

File: kyo-core/shared/src/main/scala/kyo/Barrier.scala [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                    val promise = IOPromise[Nothing, Unit]()
+                    val count   = new AtomicInteger(n)
+
+                    def await(using Frame) =
+                        IO {
+                            @tailrec def loop(c: Int): Unit < Async =
+                                if c > 0 && !count.compareAndSet(c, c - 1) then
+                                    loop(count.get)
+                                else if c == 1 then
+                                    promise.completeUnit(Result.success(()))
```
Comment (hearnadam) [inline kyo-core/shared/src/main/scala/kyo/Barrier.scala:58]: `completeUnit(Result.unit)`

File: README.md [hunk 3]
```diff
@@ -2310,6 +2310,39 @@ val d: Int < IO =
...
+// Get the number of parties still waiting
+val c: Int < IO =
+    a.map(_.pending)
+
+// Example usage with multiple fibers
+val d: Unit < Async =
+    for
+        barrier <- Barrier.init(3)
+        _       <- Async.parallel(
+                     barrier.await,
```
Comment (hearnadam) [inline README.md:2337]: I presume you would map an existing fiber to barrier.await? Might be good to show that?

**Review by hearnadam — APPROVED** [review]

File: kyo-core/shared/src/main/scala/kyo/Barrier.scala (hunk 2 above)
Comment (fwbrasil) [reply kyo-core/shared/src/main/scala/kyo/Barrier.scala:58]: I've fixed it in `Latch` as well

**Pull request merged**